src/Swarm/Game/Step.hs

```diff
@@ -1206,6 +1206,28 @@ execConst c vs s k = do
         firstOne <- findM (fmap (maybe False $ isEntityNamed name) . entityAt . (loc .+^)) sortedLocs
         return $ Out (asValue firstOne) s k
       _ -> badConst
+    Sniff -> case vs of
+      [VText name] -> do
+        firstFound <- findNearest name
+        return $ Out (asValue $ maybe (-1) fst firstFound) s k
+      _ -> badConst
+    Chirp -> case vs of
+      [VText name] -> do
+        firstFound <- findNearest name
+        mh <- use robotOrientation
+        inst <- use equippedDevices
+        let processDirection entityDir =
+              if countByName "compass" inst >= 1
+                then Just $ DAbsolute entityDir
+                else case mh >>= toDirection of
+                  Just (DAbsolute robotDir) -> Just $ DRelative $ entityDir `relativeTo` robotDir
+                  _ -> Nothing -- This may happen if the robot is facing "down"
+            val = VDir $ fromMaybe (DRelative DDown) $ do
+              entLoc <- firstFound
+              guard $ sum entLoc /= 0
```

Comment:
What is this `guard` doing?  I can't make heads or tails of it.

Reply:
But wouldn't `sum entLoc == 0` be true also for vectors like `V2 (-2) 2`?

Reply:
Also, I think `entLoc` is actually a tuple of a distance and a vector, right?  That was making me even more confused about the use of `sum`.

Reply:
Ugh.  Yes.  I think I had intended to write

    sum . fmap abs

...which is even worse.